### DISCUSSION TITLE
FluentSelect two-way binding help

### DISCUSSION THREAD

**Discussion by paulpitchford** [microsoft/fluentui-blazor#1251] [Q&A] [answered]
Hi, I'm seem to be struggling from the docs on how to enable two-way binding for a Fluent Select.

I have this model:

```
public partial class StandardsCategory
{
    public int Id { get; set; }
    public string CategoryDescription { get; set; } = null!;
}
```

I thought I could construct the select list like this:

```
<FluentSelect TOption="StandardsCategory"
              Label="Select a Category"
              Items="@Content.StandardsCategories"
              Id="category-id"
              OptionValue="@(x => x.Id)"
              @bind-Value="@Content.Standard.StandardsCategoryId" />
```

This gives these errors in the error list after compiling:
![image](https://github.com/microsoft/fluentui-blazor/assets/13196888/b5d25066-4d84-480e-bc6d-7d7c0eba411b)

Is the select list capable of it, or am I missing something? It seems like `OptionValue` only accepts type of `string` which in lots of scenarios feels a bit restrictive unless I'm completely barking up the wrong tree.

Thanks.

**Comment by vnbaaij** [comment, marked as answer]
Hi Paul, 

OptionValue is indeed always a sting (we're looking at altering that for a future version). See also #1133 for more information

I think for your example you would need to change the code for the `FluentSelect` to have:

```
OptionValue="@(x => x.Id.ToString)"
@bind-SelectedOption="@SelectedOption"
```
where `@SelectedOption` is of type `StandardsCategory`

**Reply by paulpitchford** [reply]
Thank you. The solution in #1133 got me most of the way there and then I added some code for my use case.
